app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,342 @@
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+val LIST_ITEM_HEIGHT = 56.dp
+
+val ICON_SIZE = 24.dp
+
+/**
+ * List item Composable used to display text with an optional second line of text and
+ * an optional interactable icon at the end.
+ *
+ * @param label The primary text in the item.
+ * @param description An optional second line of text.
+ * @param onClick Optional lambda for handling clicks.
+ * @param icon Optional DrawableRes used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Optional lambda for handling clicks on the icon.
+ */
+@Composable
+fun TextListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    @DrawableRes icon: Int? = null,
+    iconDescription: String? = null,
+    onIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        afterListAction = {
+            icon?.let {
+                ListItemIcon(
+                    icon = icon,
+                    iconDescription = iconDescription,
+                    onClick = onIconClick,
+                )
+
+                Spacer(modifier = Modifier.width(16.dp))
+            }
+        }
+    )
+}
+
+/**
+ * List item Composable used to display text and a [Favicon] with an optional second line of text and
+ * an optional interactable icon at the end.
+ *
+ * @param label The primary text in the item.
+ * @param description An optional second line of text.
+ * @param onClick Optional lambda for handling clicks.
+ * @param url The tab's URL.
+ * @param icon Optional DrawableRes used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Optional lambda for handling clicks on the icon.
+ */
+@Composable
+fun FavIconListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    url: String,
+    @DrawableRes icon: Int? = null,
+    iconDescription: String? = null,
+    onIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        beforeListAction = {
+            Spacer(modifier = Modifier.width(16.dp))
+
+            Favicon(
+                url = url,
+                size = ICON_SIZE,
+            )
+
+            Spacer(modifier = Modifier.width(16.dp))
```

Comment:
The before and after `Spacer` shouldn't be necessary since `ListItem` already has a horizontal padding of 16dp. I think we just need a 32dp Spacer here.

<img width="866" alt="Screen Shot 2022-04-14 at 11 40 51 AM" src="https://user-images.githubusercontent.com/1190888/163425267-85f655d0-39f2-4b03-bee4-724405b2007e.png">

Reply:
`ListItem` doesn't actually have any padding on the parent `Row` Composable, only on the internal `Column` used for text.